python.d/tor.chart.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Description: Tor Netdata python.d plugin
+# Author: Federico Ceratto <federico.ceratto@gmail.com>
+# © 2017 - released under GPLv3, see LICENSE file
+#
+# Create a configuration file at /etc/netdata/tor.conf
+#
+# local:
+#     connect_method: 'port'
+#     password: 'REPLACEME'
+#     port: 9051
+#     socket_path: '/tmp/REPLACEME'
+#
+# You can use "port" or "socket" as "connect_method"
+# The "port" and "socket_path" params will be used accodingly
+
+from bases.FrameworkServices.SocketService import SocketService
+
+try:
+    import stem
+    import stem.connection
+    import stem.control
+    ControllerError = stem.ControllerError
+    STEM_AVAILABLE = True
+except ImportError:
+    ControllerError = Exception
+    STEM_AVAILABLE = False
+
+NAME = "Tor"
+
+# default module values
+# update_every = 10
+priority = 90000
+retries = 60
+
+ORDER = ['traffic']
+CHARTS = {
+    'traffic': {
+        'options': [None, 'Tor traffic', 'bytes/s', 'Tor traffic', 'tor.traffic', 'line'],
+        'lines': [
+            ['read', 'read', 'incremental'],
+            ['write', 'write', 'incremental'],
+        ]
+    }
+}
+
+
+class Service(SocketService):
```

Comment:
btw why do you inherit from `SocketService` ?

Reply:
_parse_config is missing in SimpleService

Reply:
yes, but do you really need it? what for?